### PR TITLE
[fix] Fix error in detectEncodings when Shift-JIS, EUC-JP, etc. are specified

### PR DESCRIPTION
**Pull request by yutotnh**
## Description

By incorporating this change, specifying the following encodings in `detectEncodings` will no longer result in an error.

- SHIFT_JIS
- EUC-JP
- GB2312
- EUC-KR
- Big5
- EUC-TW

In addition, the following bugs have been fixed
- When detectEncodings is specified, the encoding is `null` when the encoding is uppercase such as SHIFT-JIS or EUC-JP.
- When minimumThreshold is set to 0, minimumThreshold is internally set to 0.2.

## Details

An error occurred when Shift-JIS was specified for detectEncodings.

### Code that generates an error

```js
    const possibleEncodings = jschardet.detectAll(fileContents, {
        detectEncodings: ["UTF-8", "windows-1252", "Shift_JIS"]
    });
```
### Error message

```text
    Encoding Shift_JIS is not supported. Supported encodings: UTF-8,UTF-32LE,UTF-32BE,UTF-32BE,UTF-16LE,UTF-16BE,X-ISO-10646-UCS-4-3412,X-ISO-10646-UCS-4-2143,HZ-GB-2312,ISO-2022-CN,ISO-2022-JP,ISO-2022-KR,windows-1251,KOI8-R,ISO-8859-5,x-mac-cyrillic,IBM866,IBM855,ISO-8859-7,windows-1253,ISO-8859-5,windows-1251,ISO-8859-2,windows-1250,TIS-620,windows-1255,windows-1255,windows-1255,windows-1252.
```

### CODE REVIEW

**Comment by yutotnh** [comment]
@aadsm 
We have confirmed that it works fine and deleted the draft status.

We understand that you are busy, but we appreciate your confirmation.

File: src/mbcsgroupprober.js [hunk 1]
```diff
@@ -47,13 +47,7 @@ function MBCSGroupProber() {
...
         new EUCTWProber()
     ];
-    const supportedCharsetNames = (function() {
-        const charsetNames = [];
-        for (const prober of this._mProbers) {
-            charsetNames.push(prober.getCharsetName())
-        }
-        return charsetNames;
-    });
+    const supportedCharsetNames = this._mProbers.map(prober => prober.getCharsetName());
```
Comment (aadsm) [inline src/mbcsgroupprober.js:50]: oh wow, thanks for catching this!

**Review by aadsm — APPROVED** [review]
Thanks for finding these bugs and submitting a fix with test cases!

**Comment by yutotnh** [comment]
Thanks for responding so quickly!

**Pull request merged**